test/UBAClientUtilities.ts

```diff
@@ -23,12 +24,47 @@ const logger = createSpyLogger().spyLogger;
 
 const chainIds = CHAIN_ID_LIST_INDICES;
 
+const tokenAddresses = chainIds.map(() => randomAddress());
+
 describe("UBAClientUtilities", function () {
+  beforeEach(async function () {
+    const [owner] = await ethers.getSigners();
+    const { configStore } = await deployConfigStore(owner, []);
+    const configStoreClient = new MockConfigStoreClient(
+      logger,
+      configStore,
+      { fromBlock: 0 },
+      UBA_MIN_CONFIG_STORE_VERSION,
+      chainIds
+    );
+    configStoreClient.setConfigStoreVersion(UBA_MIN_CONFIG_STORE_VERSION);
+    // await configStore.updateGlobalConfig(utf8ToHex(GLOBAL_CONFIG_STORE_KEYS.VERSION), UBA_MIN_CONFIG_STORE_VERSION.toString());
+    await configStoreClient.update();
+    ({ hubPool } = await hubPoolFixture());
+    const latestBlockNumber = await hubPool.provider.getBlockNumber();
+    hubPoolClient = new MockHubPoolClient(logger, hubPool, configStoreClient);
+    await hubPoolClient.update();
+    hubPoolClient.setLatestBlockNumber(latestBlockNumber);
+
+    spokePoolClients = {};
+    for (const originChainId of chainIds) {
+      const { spokePool } = await deploySpokePool(ethers);
+      const deploymentBlock = await spokePool.provider.getBlockNumber();
+
+      // Construct generic spoke pool clients with large event search configs. This should never trigger
+      // `blockRangesAreInvalidForSpokeClients` to be true.
+      const spokePoolClient = new SpokePoolClient(logger, spokePool, hubPoolClient, originChainId, deploymentBlock);
+      spokePoolClients[originChainId] = spokePoolClient;
+      await spokePoolClient.update();
+    }
```

Comment:
I'm not a huge fan of sequentially calling update. Could we update to this?
```suggestion
    for (const originChainId of chainIds) {
      const { spokePool } = await deploySpokePool(ethers);
      const deploymentBlock = await spokePool.provider.getBlockNumber();

      // Construct generic spoke pool clients with large event search configs. This should never trigger
      // `blockRangesAreInvalidForSpokeClients` to be true.
      const spokePoolClient = new SpokePoolClient(logger, spokePool, hubPoolClient, originChainId, deploymentBlock);
      spokePoolClients[originChainId] = spokePoolClient;
    }
    await Promise.all(
        Object.values(spokePoolClients).map(
            (spokePoolClient) => spokePoolClient.update()
        )
    );
```